src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -4442,13 +4445,21 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
 
             head.m_v = ASRUtils::EXPR(explicit_iter_var);
         } else {
-            body.reserve(al, x.n_body);
+            body.reserve(al, x.n_body + 2);
             head.m_v = target;
         }
 
         SymbolTable *parent_scope = current_scope;
         current_scope = al.make_new<SymbolTable>(parent_scope);
+        std::string name = "j";
```

Comment:
```py
from ltypes import i32
def main():
    i: i32
    for i in range(5):
        print(j) # not declared but this might compile
        i=i+1
        print(i)
main()
```

For that we have to mangle the name, something like `1_i_j` or `_i_j`.

Reply:
Yess it does unfortunately , do I need to use something like the following 
```
std::string j = current_scope->get_unique_name("__j");
```

Reply:
Yup, you can do that.

Reply:
Yes , I tried this ... works as expected !

Reply:
I see the example program you've shared throwing a Semantic error when I run it 
```
semantic error: Variable 'j' not declared
 --> examples/expr2.py:5:15
  |
5 |         print(j) # `j` not declared but this might compile
  |               ^ 


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that must be fixed).
```

Reply:
I think `_j` might also be caught as well, do `1_j` instead (as `1_j` is not allowed as a variable in Python)
```
from ltypes import i32
def main():
    i: i32
    for i in range(5):
        print(_j) # `_j` or `_j1` not declared but this might compile
        i=i+1
        print(i)
main()
```

Reply:
`1_for_tmp_var` or `1_for_dummy` or `1_<variable_name>_copy` or something. So that the developer or user can understand where it comes from

Reply:
Sure yeah , `j` doesn't meet the purpose completely !